test/corpus/select.txt

```diff
@@ -2016,3 +2016,66 @@ FROM
               (identifier))
             (column
               (identifier)))))))
+
+
+==================
+Where with pattern matching
+==================
+
+SELECT
+    *
+FROM
+    a
+WHERE
+        a LIKE '%a'
+    AND a NOT LIKE '%a'
+    AND a SIMILAR TO '%a'
+    AND a NOT SIMILAR TO '%a';
+
+
+---
+
+(program
+  (statement
+    (select
+      (keyword_select)
+      (select_expression
+        (all_fields)))
+    (from
+      (keyword_from)
+      (relation
+        (table_reference
+          (identifier)))
+      (where
+        (keyword_where)
+        (where_expression
+          (predicate
+            (predicate
+              (predicate
+                (predicate
+                  (field
+                    (identifier))
+                  (keyword_like)
+                  (predicate
+                    (literal)
+                    (keyword_and)
+                    (field
+                      (identifier))))
```

Comment:
Adding the node names will help make this test clearer, as it stands right now I don't think it's correct but it's hard to tell:

```suggestion
                  (predicate
                    left: (literal)
                    operator: (keyword_and)
                    right: (field (identifier))))
```

Reply:
I added the verbose node indetifiers